### ISSUE TITLE
[BUG] Error: Target page, context or browser has been closed

### ISSUE THREAD

**Issue by younesjallouli** [microsoft/playwright#28500]
**Context:**
- Playwright Version: 1.40.1
- Operating System: Mac
- Node.js version: v18.0.0
- Visual Studio Code version: 1.84.2
- Playwright for VSCode extension version: 1.0.19
- Browser: Chromium

Hello everyone,
Following the regression already signaled by some users [here](https://github.com/microsoft/playwright/issues/28388) and [here](https://github.com/microsoft/playwright/issues/28305), I was able to distill down our repo to target the bug.

The bug happens when you run the test suite with the **Show browser** checkbox enabled in the Vscode extension. The second test of the test suite will always fail

![image](https://github.com/microsoft/playwright/assets/116430697/4354f091-137e-49ce-82d5-66bdd7243089)

If I run a single test instead of the whole suite, my first run will succeed, but if I run it again, it will fail.

I've put together a small GitHub repo with 2 dummy tests. You can pull it to reproduce the issue.

**Code Snippet**
```
git clone https://github.com/younesjallouli/playwright-bug
yarn
npx playwright install --with-deps chromium
```

**Comment by mxschmitt** [comment]
I can repro.

**Comment by mxschmitt** [comment]
Investigation:
- Fixed already on ToT by https://github.com/microsoft/playwright/pull/28197
  - RANGE: https://github.com/microsoft/playwright/compare/4575c9a182b72df4d6720690ebe5b4911d240a45...738155d85df63552c065dda3abc2ae5ba62a778a
  - Good 1.41.0-alpha-nov-17-2023
  - Bad 1.40.0-alpha-nov-16-2023

**Comment by younesjallouli** [comment]
@mxschmitt will this fix also fix the situation:
- Your test opens a second tab
- You close the second tab
- Do instruction on first tab
 => Same error : Error: locator.click: Target page, context or browser has been closed

**Comment by younesjallouli** [comment]
Thank you guys 👍

**Comment by hardeep-qa** [comment]
Is a hotfix expected after this fix? Great job, team :)

**Comment by mxschmitt** [comment]
We'll release it as part of 1.41.

**Comment by liewstar** [comment]
> We recommend using out Canary release in the meantime which should have the fix applied: https://playwright.dev/docs/canary-releases我们建议在此期间使用Canary版本，该版本应应用修复程序：https://playwright.dev/docs/canary-releases
> 
> Investigation: 调查：
> 
>     * Fixed already on ToT by [fix(chromium): properly detect session closed errors for oopifs #28197](https://github.com/microsoft/playwright/pull/28197)已通过修复（铬）修复ToT：正确检测oopifs的会话关闭错误#28197
>       
>       * RANGE: [4575c9a...738155d](https://github.com/microsoft/playwright/compare/4575c9a182b72df4d6720690ebe5b4911d240a45...738155d85df63552c065dda3abc2ae5ba62a778a) 范围： 4575c9a...738155d
>       * Good 1.41.0-alpha-nov-17-2023良好1.41.0-alpha-nov-17-2023
>       * Bad 1.40.0-alpha-nov-16-2023坏1.40.0-α-十一月16-2023

Is there currently a way to avoid or solve this problem with Python's playwright？

**Comment by mxschmitt** [comment]
Unfortunately not, we recommend waiting for 1.41 or sticking with an older version of Playwright in the meantime.

**Comment by bjerkins** [comment]
@younesjallouli have you tried the newly dropped 1.41 ? The issue is still there as far as I can see ... 🤔

**Comment by mxschmitt** [comment]
I tried the repro from https://github.com/microsoft/playwright/issues/28500#issue-2026979015 and it failed for me on 1.40 and passed on 1.41. @bjerkins could you file a separate issue please? Most likely your issue is different then.

**Comment by Valentyna** [comment]
I've updated to `playwright:v1.41.1` and I'm still seeing error:
```
Error: locator.click: Target page, context or browser has been closed
```

**Comment by mxschmitt** [comment]
Could you please file a separate issue with a reproduction? Like steps for us in order to replicate the issue you are encountering. This would help a ton in order for us to triage this issue further, thanks!

**Comment by bjerkins** [comment]
> Could you please file a separate issue with a reproduction? Like steps for us in order to replicate the issue you are encountering. This would help a ton in order for us to triage this issue further, thanks!

@mxschmitt , I can confirm that @younesjallouli 's small example does indeed work properly when using the latest version. However, for some reason our setup still has problems and I haven't had time to pinpoint what exactly causes it. Isn't there some logs we can provide ?

**Comment by coexJoy** [comment]
The issue persists on the latest v`1.42.1`. I am running the playwright with Cucumber v`10.0.1`. I still can't figure out why the browser is getting closed off in between scenarios.

The funny thing is, it works fine for the first two scenarios and fails for the last one out of three. And the order of scenarios doesn't matter either. So I am guessing it has nothing to do with the scenarios itself.

**Comment by gallak-openfin** [comment]
I recently ran into this and found that it was due to re-using `page` handlers accidentally.

We have a complex page and I had created a caching model that lived outside of the fixture scope itself. Presumably this lead to caching locators created with the `page` fixture from other tests. Since most of our tests have similar top-level seed data it was hitting the cache.

Creating the cache at the fixture scope solved this for me.
